tests/meta-mender-toradex-nxp-ci/recipes-bsp/u-boot/files/colibri-imx7/fw_env.config

```diff
@@ -0,0 +1,10 @@
+# Configuration file for fw_(printenv/setenv) utility.
+# Up to two entries are valid, in this case the redundant
+# environment sector is assumed present.
+# Notice, that the "Number of sectors" is not required on NOR and SPI-dataflash.
+# Futhermore, if the Flash sector size is ommitted, this value is assumed to
+# be the same as the Environment size, which is valid for NOR and SPI-dataflash
+
+# MTD device name       Device offset   Env. size       Flash sector size       Number of sectors
+/dev/mtd3               0x00000000      0x00020000      0x20000                 4
+/dev/mtd3               0x00020000      0x00020000      0x20000                 4
```

Comment:
Above is not completely correct. 

The field "Number of sectors" indicates how many NAND sectors to occupy for the U-boot ENV, in the above case 0x80000 (512 kB) will be used for each part. This must match the CONFIG_ENV_RANGE variable in the U-boot conf (which I believe is also setup to 512 kB in colibri-imx7.h).  CONFIG_ENV_RANGE is used so that it can tolerate bad blocks in the ENV area.

Above will result in that the "second" environment will be overwritten in case of bad blocks on the "first" environment part.

One possible solution is to increase the size of the u-boot-env MTD part (1MB) and change the device offset of the second env area to 0x80000

or

decrease the number of sectors to 2 (256 kB) both in fw_env.config and in u-boot conf, but you still need to adjust the device offset for the second area but you then do not change the size of u-boot-env MTD part.

or 

one can omit the "Number of sectors" field and CONFIG_ENV_RANGE from u-boot, then it will only occupy CONFIG_ENV_SIZE.